### PR TITLE
Fragments

### PR DESCRIPTION
**Pull request by librasteve**
This request is an implementation of Fragments for the Cro Template language.

It addresses this Feature Request Issue https://github.com/croservices/cro-webapp/issues/90 which hopefully provides a rationale for the new feature.

I will shortly make a PR for cro-website to update the docs to include the new feature.

I suggest a point bump in the major version since this extends the API, e.g. to `0.9.0` which I leave to the person merging this PR if it is approved.

### CODE REVIEW

**Comment by patrickbkr** [comment]
Thanks for putting work into this. This is looking promising. I'll give this a review as soon as I find some time.

File: lib/Cro/WebApp/Template/Repository.rakumod
Comment (patrickbkr) [inline lib/Cro/WebApp/Template/Repository.rakumod:35]: Debug leftover.

File: lib/Cro/WebApp/Template/Repository.rakumod
Comment (patrickbkr) [inline lib/Cro/WebApp/Template/Repository.rakumod:33]: Can you explain, why the `my $*TEMPLATE-REPOSITORY = $!repository;` isn't needed in this case?

File: lib/Cro/WebApp/Template/AST.rakumod
Comment (patrickbkr) [inline lib/Cro/WebApp/Template/AST.rakumod:21]: Is this dead code or missing the implementation?

**Review by patrickbkr — CHANGES_REQUESTED** [review]
All in all looking good. Just some polishing is missing.

File: cro-webapp.iml
Comment (librasteve) [reply cro-webapp.iml]: I think best practice is to not have comma files .idea/* and *.iml in the repo - deleted from the PR

File: lib/Cro/WebApp/Template/AST.rakumod
Comment (librasteve) [reply lib/Cro/WebApp/Template/AST.rakumod:21]: good catch - this is indeed dead code - compile now just makes both the fragment and the whole - and then the render phase picks out the fragment if needed at method render (Repository.rakumod ll31-38)

File: lib/Cro/WebApp/Template/Repository.rakumod
Comment (librasteve) [reply lib/Cro/WebApp/Template/Repository.rakumod:33]: I am pretty sure that the compile phase loads the template `$.exports` on `Cro::WebApp::Template::Compiled.new` and that this attribute does not need adjusting according to the current`$!repository` whereas the `&.renderer` needs the repository path to pick the correct precompiled template.

That said, this statement is a little bit weak ... I can imagine that future evolutions of the code may expect both the exports and the repository attributes to have the same dependencies (especially in regard to global repository selectors) and that the current design is weak in this area. For example `$.exports` may one day be changed to a method, not a public attribute.

Therefore I have adjusted the code to put both arms of the if / else into the same global situation.

**Comment by librasteve** [comment]
Good catches in there - thanks!

Hopefully all is now shiny.

I will now go change § to ~ in the docs ;-)

**Comment by patrickbkr** [comment]
@librasteve: Are you aware that the Cro project is currently severely lacking maintainers? Currently I and pe_elle (who only recently joined) are the only ones doing maintainer work (currently reviewing is what's needed the most). Given there is a policy of one-review-before-merge, my own PRs are currently effectively frozen because I can't review my own PRs. Would you consider joining the team? It does seem like you know the stack.

**Comment by librasteve** [comment]
yes - I would be very happy to join the team - thank you for the invitation

**Comment by patrickbkr** [comment]
I've just sent you an invite. How can you be reached? I guess this github issue is not the best communication platform. :-)

**Review by patrickbkr — APPROVED** [review]
Looking good!

**Comment by patrickbkr** [comment]
Are you OK with merging this as well now?

**Comment by librasteve** [comment]
OK - I gather that CI was failing before my PR

**Comment by patrickbkr** [comment]
Yeah, I know what's wrong with the CI. Fix is upcoming.

**Pull request merged**
